src/Courses/CourseInterfaces.tsx

```diff
@@ -96,10 +107,15 @@ export class NewCourseTopicObj extends TopicObject {
     endDate: Date = new Date();
     deadDate: Date = new Date();
     partialExtend: boolean = false;
+    studentTopicOverride: any[] = [];
 
     public constructor(init?:Partial<NewCourseTopicObj>) {
-        super();
+        super(init);
         Object.assign(this, init);
+        
+        if (!_.isNull(init?.questions)) {
```

Comment:
Why do we have to do this? is it for the default values?

Reply:
This applies the typing/constructor logic to all the subobjects that come back from the API.

Reply:
Gotcha you mean for instance of, I wonder if there is a more efficient way of assigning a prototype 🤔